tests/functions/module_lookup.test.ts

```diff
@@ -1305,3 +1305,163 @@ describe("INDEX formula", () => {
     expect(evaluateCellFormat("B4", { B4: "=INDEX(A1:B2, 1, 2)", ...grid })).toBe("0%");
   });
 });
+
+describe("INDIRECT formula", () => {
+  test("Check argument validity", () => {
+    expect(evaluateCell("A1", { A1: "=INDIRECT()" })).toBe("#BAD_EXPR");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("B1", "string")' })).toBe("#ERROR");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("B1", false)' })).toBe("#REF");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("R1C1", true)' })).toBe("#REF");
+    expect(evaluateCell("A1", { A1: '=INDIRECT("wrong_reference")' })).toBe("#REF");
+    expect(evaluateCell("A1", { A1: "=INDIRECT(,false)" })).toBe("#ERROR");
+  });
+
+  test("Using address string as reference (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "A2",
+      A4: '=INDIRECT("A1")',
+      A5: '=INDIRECT("A2")',
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe("A2");
+  });
+
+  test("Using cell content as reference (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "A1",
+      A3: "A2",
+      A4: "=INDIRECT(A2)",
+      A5: "=INDIRECT(A3)",
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe("A1");
+  });
+
+  test("Using computation result as reference (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "A2",
+      A4: '=INDIRECT("A"&ROW(B1))',
+      A5: '=INDIRECT(CONCAT("A","2"))',
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe("A2");
+  });
+
+  test("Reference to a cell as a function argument (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "2",
+      A3: "3",
+      A4: '=MAX(INDIRECT("A1"),0)',
+      A5: '=INDIRECT("A2")*INDIRECT("A3")',
+    });
+    expect(grid.A4).toBe(1);
+    expect(grid.A5).toBe(6);
+  });
+
+  test("Reference to a range as a function argument (A1 notation)", () => {
+    const grid = evaluateGrid({
+      A1: "1",
+      A2: "2",
+      A3: "3",
+      A4: '=SUM(INDIRECT("A1:A3"))',
+    });
+    expect(grid.A4).toBe(6);
+  });
+
+  test("Reference to a cell and range of a different sheet (A1 notation)", () => {
+    const model = new Model();
+    const sheetId = model.getters.getActiveSheetId();
+    createSheet(model, { sheetId: "sheet2", activate: true });
+    setCellContent(model, "A1", "1");
+    setCellContent(model, "A2", "2");
+    activateSheet(model, sheetId);
+    setCellContent(model, "A1", '=INDIRECT("sheet2!A1")');
+    setCellContent(model, "A2", '=SUM(INDIRECT("sheet2!A1:A2"))');
+    expect(getCellContent(model, "A1")).toBe("1");
+    expect(getCellContent(model, "A2")).toBe("3");
+  });
+
+  test("Using address string as reference (RC notation)", () => {
```

Comment:
might be worth doing test.each(AIReference, RCReference) if it's not too annoying